File: 슬라이딩 윈도우/프로그래머스Lv2-연속 부분 수열 합의 개수.py
def solution(elements):
    numbers = elements * 2
    
    result = set()
    
    for size in range(1, len(elements) + 1):
        for i in range(len(elements)):
            result.add(sum(numbers[i:i + size]))
            
    return len(result)
"""
정확성  테스트
테스트 1 〉	통과 (0.01ms, 10.2MB)
테스트 2 〉	통과 (46.44ms, 13.1MB)
테스트 3 〉	통과 (136.19ms, 14.2MB)
테스트 4 〉	통과 (301.04ms, 18.3MB)
테스트 5 〉	통과 (532.12ms, 26.9MB)
테스트 6 〉	통과 (1013.81ms, 27MB)
테스트 7 〉	통과 (1434.92ms, 26.9MB)
테스트 8 〉	통과 (2060.03ms, 27.8MB)
테스트 9 〉	통과 (2898.86ms, 43.7MB)
테스트 10 〉	통과 (3736.48ms, 43.6MB)
테스트 11 〉	통과 (692.66ms, 27MB)
테스트 12 〉	통과 (829.26ms, 26.9MB)
테스트 13 〉	통과 (1312.15ms, 26.9MB)
테스트 14 〉	통과 (1271.97ms, 27MB)
테스트 15 〉	통과 (1536.19ms, 26.9MB)
테스트 16 〉	통과 (1847.47ms, 43.8MB)
테스트 17 〉	통과 (2153.33ms, 43.8MB)
테스트 18 〉	통과 (2535.09ms, 43.8MB)
테스트 19 〉	통과 (3081.87ms, 43.6MB)
테스트 20 〉	통과 (3599.99ms, 43.7MB)
"""

File: 슬라이딩 윈도우/프로그래머스Lv2-연속 부분 수열 합의 개수.js
const sum = (arr) => arr.reduce((a, b) => a + b, 0)

function solution(elements) {
  const numbers = elements.concat(elements)

  const result = new Set()

  for (let size = 1; size < elements.length + 1; size++) {
    for (let i = 0; i < elements.length; i++) {
      result.add(sum(numbers.slice(i, i + size)))
    }
  }

  return [...result].length
}
// 정확성  테스트
// 테스트 1 〉	통과 (0.13ms, 33.5MB)
// 테스트 2 〉	통과 (18.86ms, 39.4MB)
// 테스트 3 〉	통과 (37.87ms, 42.4MB)
// 테스트 4 〉	통과 (76.99ms, 42.5MB)
// 테스트 5 〉	통과 (126.09ms, 46.6MB)
// 테스트 6 〉	통과 (209.74ms, 46.6MB)
// 테스트 7 〉	통과 (338.81ms, 52.1MB)
// 테스트 8 〉	통과 (458.26ms, 51.8MB)
// 테스트 9 〉	통과 (611.47ms, 51.8MB)
// 테스트 10 〉	통과 (834.17ms, 53MB)
// 테스트 11 〉	통과 (164.80ms, 46.8MB)
// 테스트 12 〉	통과 (199.83ms, 46.6MB)
// 테스트 13 〉	통과 (247.13ms, 46.6MB)
// 테스트 14 〉	통과 (335.89ms, 51.8MB)
// 테스트 15 〉	통과 (397.66ms, 51.9MB)
// 테스트 16 〉	통과 (443.83ms, 51.8MB)
// 테스트 17 〉	통과 (538.65ms, 52.8MB)
// 테스트 18 〉	통과 (648.57ms, 51.9MB)
// 테스트 19 〉	통과 (759.79ms, 51.8MB)
// 테스트 20 〉	통과 (840.55ms, 53MB)
